app/backend/src/clj/teet/cooperation/cooperation_db.clj

```diff
@@ -144,6 +144,20 @@
   (get-in (du/entity db application-id)
     [:cooperation.3rd-party/_applications 0 :cooperation.3rd-party/name]))
 
+(defn application-3rd-party-uuid [db application-id]
+  (let [name-of-3rd-party (application-3rd-party db application-id)]
+    (d/q '[:find ?id
+           :in $ ?name
+           :keys uuid
+           :where [?cooperation :cooperation.3rd-party/name ?name]
```

Comment:
why search 3rd party by name? you can directly navigate `:cooperation.3rd-party/applications` attribute to get the 3rd party